pkg/hive/cell/module.go

```diff
@@ -56,16 +63,31 @@ func (m *module) logger(log logrus.FieldLogger) logrus.FieldLogger {
 	return log.WithField(logfields.LogSubsys, m.id)
 }
 
-func (m *module) moduleScopedStatusReporter(p Health) HealthReporter {
-	return p.forModule(m.id)
+func (m *module) moduleID() ModuleID {
+	return ModuleID(m.id)
+}
+
+func (m *module) fullModuleID(parent FullModuleID) FullModuleID {
+	if parent == "" {
```

Comment:
We should hang ctors on the type and let the type encapsulate how this is represented internally vs exposing the semantic at the call site?

Reply:
Yep, will clean up.